src/main.rs

```diff
@@ -287,9 +288,29 @@ const APP: () = {
             &ccdr.clocks,
         );
 
-        let mut sampling_timer = sampling_timer::SamplingTimer::new(timer2);
+        let mut sampling_timer = timers::SamplingTimer::new(timer2);
         let sampling_timer_channels = sampling_timer.channels();
 
+        let mut timestamp_timer = {
+            // The timer frequency is manually adjusted below, so the 1KHz setting here is a
+            // dont-care.
+            let timer5 =
+                dp.TIM5.timer(1.khz(), ccdr.peripheral.TIM5, &ccdr.clocks);
+
+            // The time stamp timer must run at exactly a multiple of the sample timer based on the
+            // batch size. To accomodate this, we manually set the period identical to the sample
+            // timer, but use a prescaler that is `BATCH_SIZE` longer.
+            let mut timer = timers::TimestampTimer::new(timer5);
+            timer.set_period(sampling_timer.get_period());
+            timer.set_prescaler(
+                sampling_timer.get_prescaler() * SAMPLE_BUFFER_SIZE as u16,
```

Comment:
The prescaler decreases resolution. It should be as low as possible. Instead increase the period.
Or is that not feasible?

Reply:
This was some legacy implementation before our conversation today - it's also no longer present.